src/index.js

```diff
@@ -43,10 +44,10 @@ window.geckoProfilerPromise = new Promise(function(resolve) {
 });
 
 const store = createStore();
+setupUrlHandling(store);
```

Comment:
This becomes a synchronous step before mounting the React components. I re-organized these steps to make it more tidy.